### ISSUE TITLE
sqrt_with_context() doesn't respect rounding mode?

### ISSUE THREAD

**Issue by jorisguex** [akubera/bigdecimal-rs#133]
The following code:

```rust
fn main() {
    let val = bigdecimal::BigDecimal::new(225.into(), 2);
    let prec = std::num::NonZero::new(1).unwrap();
    println!("sqrt({val}) with prec={prec}:");
    println!("Actual: {}", val.sqrt().unwrap());
    for rounding_mode in [bigdecimal::RoundingMode::Up, bigdecimal::RoundingMode::Down] {
        let res = val.sqrt_with_context(&bigdecimal::Context::new(prec, rounding_mode)).unwrap();
        println!("{rounding_mode:?}: {res}");
    }
}
```

produces the following output:

```
sqrt(2.25) with prec=1:
Actual: 1.500000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
Up: 1
Down: 1
```

I would have expected that when using a `Context` with `RoundingMode::Up`, the output value would be 2 instead of 1. The docs say "Take the square root of the number, using context for precision and rounding", so it seems like this is either a bug or I am doing something wrong?

**Comment by akubera** [comment]
Thanks for reporting it. The sqrt implementation is only rounding down (truncating) and only uses the context to set the precision.

I think I'll have time to check this next weekend.

**Comment by danilopedraza** [comment]
It seems like this was fixed in [8d38646](https://github.com/akubera/bigdecimal-rs/commit/8d38646bfacdf1480cfce86bb8fe569fb19547f4), [ed6cfc3](https://github.com/akubera/bigdecimal-rs/commit/ed6cfc3c6046c71cce15fb23281dfebdd1c2771b), and [fcdf76b
](https://github.com/akubera/bigdecimal-rs/commit/fcdf76b34ee0ad52f9f7a853b8d30553a9a95f5e)

Should we close this?

**Comment by akubera** [comment]
Yup
